### PR TITLE
Fix externalips overridden

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
make gateway service's externalIPs takes precedence.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -392,7 +392,11 @@ func (c *Controller) onServiceEvent(curr interface{}, event model.Event) error {
 		delete(c.networkGateways, svcConv.Hostname)
 		c.Unlock()
 	default:
-		if isNodePortGatewayService(svc) {
+		// First, process nodePort gateway service, whose externalIPs specified
+		// and loadbalancer gateway service
+		if svcConv.Attributes.ClusterExternalAddresses != nil {
```
Comment (stevenctl) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:397]: Will this make it so we never update the service node port address after we get the first one?

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:397]: if the externalIPs is specified, then nodeport is not respected. This is to make this a determined behavior.

Comment (stevenctl) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:397]: I didn't catch that `svcConv` was freshly generated here, I thought externalIPs may have already been populated by NodePort on a previous update. 

One other question: When using the external IPs we won't map the node port to the service port. Do we expect that services with type NodePort will also be configured to expose the Service port (usually 15443)?

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:397]: Yes, in this case remote cluster's gateway can be accessed by externalIP:15443

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:397]: @stevenctl Any other concern

**Review by stevenctl — APPROVED** [review]
LGTM

File: pilot/pkg/serviceregistry/kube/controller/controller.go (hunk 1 above)
Comment (stevenctl) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:397]: No, this looks good. Thanks for the clarifications!

**Pull request merged**